pubsub-mapped-api/src/main/java/com/google/pubsub/clients/consumer/ack/MessageDispatcher.java

```diff
@@ -631,20 +624,22 @@ private void extendAckDeadlines(List<PendingModifyAckDeadline> ackDeadlineExtens
     List<PendingModifyAckDeadline> modifyAckDeadlinesToSend =
         Lists.newArrayList(ackDeadlineExtensions);
     PendingModifyAckDeadline nacksToSend = new PendingModifyAckDeadline(0);
+
+    Set<String> acksToExtend;
+
     synchronized (pendingNacks) {
-      if (!pendingNacks.isEmpty()) {
-        try {
-          for (String ackId : pendingNacks) {
-            nacksToSend.addAckId(ackId);
-          }
-          logger.log(Level.FINER, "Sending {0} nacks", pendingNacks.size());
-        } finally {
-          pendingNacks.clear();
-        }
-        modifyAckDeadlinesToSend.add(nacksToSend);
+      acksToExtend = pendingNacks;
+      pendingNacks = ConcurrentHashMap.newKeySet();
```

Comment:
what if sendAckOperations throws an error? are acksToSend put back in pendingNacks, or are they lost?

Reply:
I don't understand what was happening before, I mean pendingNacks was being cleared anyway, right? If it's the case, then I am doing the same. I think I didn't change the logic here - or I understand the try - finally wrong then.